src/ray/object_manager/object_manager.cc

```diff
@@ -94,6 +116,7 @@ void ObjectManager::NotifyDirectoryObjectDeleted(const ObjectID &object_id) {
   RAY_CHECK(it != local_objects_.end());
   local_objects_.erase(it);
   ray::Status status = object_directory_->ReportObjectRemoved(object_id, client_id_);
```

Comment:
We should make sure to only do this if the object was not inlined. Otherwise, we will add an eviction entry to the GCS for inlined objects.